elixir/apps/domain/test/domain/policies_test.exs

```diff
@@ -0,0 +1,371 @@
+    end
+
+    test "returns all policies for account admin subject", %{account: account} do
+      actor = ActorsFixtures.create_actor(type: :account_user, account: account)
+      identity = AuthFixtures.create_identity(account: account, actor: actor)
+      subject = AuthFixtures.create_subject(identity)
+
+      PoliciesFixtures.create_policy(account: account)
+      PoliciesFixtures.create_policy(account: account)
+      PoliciesFixtures.create_policy()
+
+      assert {:ok, policies} = list_policies(subject)
+      assert length(policies) == 2
+    end
+
+    test "returns error when subject has no permission to view policies", %{subject: subject} do
+      subject = AuthFixtures.remove_permissions(subject)
+
+      assert list_policies(subject) ==
+               {:error,
+                {:unauthorized,
+                 [
+                   missing_permissions: [
+                     {:one_of,
+                      [
+                        Policies.Authorizer.manage_policies_permission(),
+                        Policies.Authorizer.view_available_policies_permission()
+                      ]}
+                   ]
+                 ]}}
+    end
+  end
+
+  describe "create_policy/2" do
+    test "returns changeset error on empty params", %{subject: subject} do
+      assert {:error, changeset} = create_policy(%{}, subject)
+
+      assert changeset.errors == [
+               name: {"can't be blank", [validation: :required]},
+               actor_group_id: {"can't be blank", [validation: :required]},
+               resource_id: {"can't be blank", [validation: :required]}
+             ]
+    end
+
+    test "returns changeset error on invalid params", %{subject: subject} do
+      assert {:error, changeset} =
+               create_policy(
+                 %{name: 1, actor_group_id: "foo", resource_id: "bar"},
+                 subject
+               )
+
+      assert changeset.errors == [
+               name: {"is invalid", [{:type, :string}, {:validation, :cast}]}
+             ]
+    end
+
+    test "returns error when subject has no permission to manage policies", %{subject: subject} do
+      subject = AuthFixtures.remove_permissions(subject)
+
+      assert create_policy(%{}, subject) ==
+               {:error,
+                {:unauthorized,
+                 [
+                   missing_permissions: [
+                     {:one_of, [Policies.Authorizer.manage_policies_permission()]}
+                   ]
+                 ]}}
+    end
+
+    test "returns changeset error when trying to create policy with another account actor_group",
+         %{
+           account: account,
+           subject: subject
+         } do
+      other_account = AccountsFixtures.create_account()
+
+      resource = ResourcesFixtures.create_resource(account: account)
+      other_actor_group = ActorsFixtures.create_group(account: other_account)
+
+      assert {:error, changeset} =
+               create_policy(
+                 %{
```

Comment:
Just for the sake of readabiluty it's better to move large maps out to a variable, eg:
```
attrs = %{...}
assert {:error, changeset} = create_polcity(attrs, subject)
```

Reply:
Sounds good 👍

Reply:
This has been updated